conman/routes/tests/test_handlers.py

```diff
@@ -101,40 +100,91 @@ def test_handle_no_url_match(self):
         self.assertFalse(dummy_view.called)
 
 
+class URLConfHandlerCheckTest(TestCase):
+    """Tests for URLConfHandler.check()."""
+    def test_no_urlconf(self):
+        """When the route has no urlconf, return an error."""
+        class RouteWithNoURLConf(Route):
+            handler_class = URLConfHandler
+
+        errors = URLConfHandler.check(RouteWithNoURLConf)
+        expected = Error(
+            'RouteWithNoURLConf must have a `urlconf` attribute.',
+            hint=(
+                'The urlconf must be a dotted path. ' +
+                'This is a requirement of URLConfHandler.'
+            ),
+            obj=RouteWithNoURLConf,
+        )
+        self.assertEqual(errors, [expected])
+
+    def test_has_urlconf(self):
+        """When the Route has a urlconf, all's well."""
+        class RouteWithURLConf(Route):
+            handler_class = URLConfHandler
+            urlconf = 'a.dotted.path'
+
+        errors = URLConfHandler.check(RouteWithURLConf)
+        self.assertEqual(errors, [])
+
+
+class RouteViewHandlerCheckTest(TestCase):
+    """Tests for RouteViewHandler.check()."""
+    def test_no_view(self):
+        """When the route has no view, return an error."""
+        class RouteWithNoView(Route):
+            handler_class = RouteViewHandler
+
+        errors = RouteViewHandler.check(RouteWithNoView)
+        expected = Error(
+            'RouteWithNoView must have a `view` attribute.',
+            hint='This is a requirement of RouteViewHandler.',
+            obj=RouteWithNoView,
+        )
+        self.assertEqual(errors, [expected])
+
+    def test_function(self):
+        """When the Route has a view function, all's well."""
+        class RouteWithView(Route):
+            handler_class = RouteViewHandler
+
+            def view(request):
+                return
+
+        errors = RouteViewHandler.check(RouteWithView)
+        self.assertEqual(errors, [])
+
+
 class RouteViewHandlerHandleTest(TestCase):
     """Test RouteViewHandler.handle()."""
     def setUp(self):
         """Create a route, request, and view for use in this test."""
         class MockViewRoute(Route):
-            view = mock.MagicMock()
+            view = staticmethod(mock.MagicMock())
```

Comment:
For some reason, this still fails without the `staticmethod`. Perhaps a peculiarity of `mock`?

Given the warnings we're getting about this model being re-defined, I suspect I'm doing things pretty hideously here, but can't quite see the solution at this time of night.